src/app.rs

```diff
@@ -307,10 +308,133 @@ impl App {
             }
             KeyCode::Esc => self.reset_message_selection(),
             KeyCode::Char(c) => self.put_char(c),
+            KeyCode::Tab => {
+                if let Some(idx) = self.data.channels.state.selected() {
+                    self.add_reaction(idx, self.data.input.is_empty()).await;
+                }
+            }
             _ => {}
         }
     }
 
+    pub async fn add_reaction(&mut self, channel_idx: usize, remove: bool) -> Option<()> {
+        let input: String = self.data.input.drain(..).collect();
+        let reaction = self.take_emoji(&input);
+        log::error!("\"{:?}\"", reaction);
```

Comment:
I guess this is a left over from debugging, right?

Reply:
Oh right! I broke the whole mechanism when reformatting so I had to use some heavy debugging.